### PR TITLE
Add sharing of attachments.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk 

img_6918
img_6917
img_6916
img_6915

### CODE REVIEW

File: Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m [hunk 1]
```diff
@@ -271,6 +277,29 @@ - (void)performEditingAction:(SEL)action
...
+        return nil;
+    }
+
+    TSMessage *message = (TSMessage *)self.interaction;
+
+    if (![message hasAttachments]) {
+        return nil;
+    }
+    OWSAssert(message.attachmentIds.count <= 1);
+    NSString *attachmentID = message.attachmentIds[0];
```
Comment (charlesmchen) [inline Signal/src/Models/TSMessageAdapaters/TSMessageAdapter.m]: I've made the assumption that messages have at most one attachment.

**Comment by charlesmchen** [comment]
Fixes: https://github.com/WhisperSystems/Signal-iOS/issues/1747

File: Signal/translations/en.lproj/Localizable.strings [hunk 2]
```diff
@@ -169,6 +169,9 @@
 /* Short name for edit menu item to save contents of media message. */
 "EDIT_ITEM_SAVE_ACTION" = "Save";
 
+/* Short name for edit menu item to share contents of media message. */
```
Comment (michaelkirk) [inline Signal/translations/en.lproj/Localizable.strings]: nit: Do you want to re-use the existing "ATTACHMENT_VIEW_SHARE_ACTION" which has an identical translation and  description?

Comment (charlesmchen) [inline Signal/translations/en.lproj/Localizable.strings]: Sure. I've de-duplicated other strings too.

**Pull request merged**